src/lib/services/notes/update.ts

```diff
@@ -0,0 +1,13 @@
+import { db } from '@/lib/db'
+import { z } from 'zod'
+
+export const updateNoteInput = z.object({
+  id: z.string(),
+  color: z.string().optional(),
+  content: z.string().optional(),
+})
```

Comment:
Move this to a separate file, like `src/lib/schemas/update-note.ts` so that we can also use it to validate forms on the client. I'd do something like this:

```ts
// src/lib/schemas/update-note.ts
import { z } from 'zod'
export const updateNoteInput = z.object({
  id: z.string(),
  color: z.string().optional(),
  content: z.string().optional(),
})

export type UpdateNoteData = z.infer<typeof updateNoteInput>

// src/lib/services/notes/update.ts
import type { UpdateNoteData } from '@/lib/schemas/update-note'

export async function updateNote(data: UpdateNoteData) { ... }
```

While you could also import the schema to validate the data being passed to the function again, I generally delegate validation to controllers, since they receive the user input, and then pass in the correct input to services, since I have control of the backend transformations.

The motivation for having a dedicated schema file is because the schema doesn't contain any sensitive code, like database connections, environment variables, etc. and can be used on either the server or the client. And we __do__ want to use them in both places: on the server to validate incoming data, and on the client to validate form data.